ui/hooks/useAssetDetails.js

```diff
@@ -38,6 +39,21 @@ export function useAssetDetails(tokenAddress, userAddress, transactionData) {
         transactionData,
         nfts,
       );
+      const token = util.getContractAtAddress(tokenAddress);
+
+      const symbol = await token.symbol();
+      const name = await token.name();
+
+      if (!assetDetails.symbol) {
+        setCurrentAsset({
+          ...assetDetails,
+          symbol: symbol[0],
+          name: name[0],
+        });
+        dispatch(hideLoadingIndication());
+        return;
+      }
```

Comment:
```suggestion
      if (!assetDetails.symbol) {
        assetDetails = {
          ...assetDetails,
          symbol: symbol[0],
          name: name[0],
        };
      }
```

Since the existing code already calls `setCurrentAsset` and `dispatch(hideLoadingIndication())`, it seems simpler to just reassign `assetDetails` and let both paths share the final code.